### PR TITLE
Try new way of enabling release events.

### PR DESCRIPTION
**Pull request by mozbhearsum**
Using the new event talked about in https://docs.taskcluster.net/manual/integrations/github.

### CODE REVIEW

**Comment by mozbhearsum** [comment]
It should be noted that this isn't possible to test in a PR properly, so I've added a completely new Task that only responsd to release events. My intention is to try it out for the next push (probably next week) and then remove the old Docker build tasks if the new ones work well.

**Comment by mozbhearsum** [comment]
@owlishDeveloper, can you have a look at this? @rail, maybe you too? I don't think @owlishDeveloper will be to approve this because of group permissions.,

File: .taskcluster.yml [hunk 1]
```diff
@@ -155,11 +154,62 @@ tasks:
...
+    payload:
+      maxRunTime: 3600
+      image: "taskcluster/image_builder:0.1.3"
+      features:
+        dind: true
+        taskclusterProxy: true
+      command:
+        - "/bin/bash"
+        - "-c"
+        - "git clone {{ event.head.repo.url }} && cd balrog && git checkout {{ event.version }} && scripts/push-dockerimage.sh"
```
Comment (owlishDeveloper) [inline .taskcluster.yml]: Correct clone in this case will be: git clone -b {{ event.version }} {{ event.head.repo.url }} repo

**Comment by owlishDeveloper** [comment]
@mozbhearsum Except for that clone thing looks good to me!

**Comment by mozbhearsum** [comment]
Hmm, when I tried it locally, "git clone -b v2.16 https://github.com/mozilla/balrog" cloned the repo and put it on the revision I expected? Unless I'm missing something, I don't think the explicit checkout does anything?

```
(balrog) ➜  a git clone -b v2.16 https://github.com/mozilla/balrog                                              
Cloning into 'balrog'...
remote: Counting objects: 10181, done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 10181 (delta 8), reused 0 (delta 0), pack-reused 10148
Receiving objects: 100% (10181/10181), 59.81 MiB | 3.25 MiB/s, done.
Resolving deltas: 100% (5549/5549), done.
Checking connectivity... done.
Note: checking out '4d4d7baae8834074cc37429284e9572d622d11ac'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

(balrog) ➜  a cd balrog
(balrog) ➜  balrog git:(4d4d7ba) git checkout v2.16                                  
HEAD is now at 4d4d7ba... Version bump for next release.
(balrog) ➜  balrog git:(4d4d7ba) 

```

**Comment by owlishDeveloper** [comment]
@mozbhearsum  you're right! Your first and second versions are correct, and they do the same thing! Thank you for correcting me, I'll update the docs. Sorry for the confusion!

**Pull request merged**